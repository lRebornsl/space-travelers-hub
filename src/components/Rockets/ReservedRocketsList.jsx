import { useSelector } from 'react-redux';

const ReservedRocketsList = () => {
  const rockets = useSelector((state) => state.rocketsSlice.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div>
      <h2>My Rockets</h2>
      {reservedRockets.length === 0 ? (
        <p>No rockets reserved yet.</p>
      ) : (
        <ul>
          {reservedRockets.map((rocket) => (
            <li key={rocket.id}>
              {rocket.name}
            </li>
          ))}
        </ul>
      )}
    </div>
  )
}

export default ReservedRocketsList
