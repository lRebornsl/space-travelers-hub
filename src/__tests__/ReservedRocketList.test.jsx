import { render, waitFor } from '@testing-library/react';
import ReservedRocketsList from '../components/Rockets/ReservedRocketsList';
import { beforeEach, describe, it, expect } from 'vitest';
import { Provider } from 'react-redux';
import store from '../redux/Store';

describe('Active Missions List', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <ReservedRocketsList />
      </Provider>
    );
  });

  it('ReservedRocketsList renders correctly', async () => {
    await waitFor(() => expect(ReservedRocketsList).toMatchSnapshot());
  });

  it('Should show "no active missions"', async () => {
    const { findAllByText } = render(
      <Provider store={store}>
        <ReservedRocketsList />
      </Provider>
    );
    await waitFor(() => expect(findAllByText('No active missions yet.')).toBeTruthy());
  });
});