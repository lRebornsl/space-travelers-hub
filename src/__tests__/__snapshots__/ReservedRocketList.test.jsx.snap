// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Active Missions List > ReservedRocketsList renders correctly 1`] = `[Function]`;
